### PR TITLE
created cache decorator

### PR DESCRIPTION
**Pull request by Timur5050**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_data = {}
+
+    def wrapper(*args) -> any:
```
Comment (vsmutok) [inline app/main.py]: use `functools.wraps`

File: app/main.py [hunk 2]
```diff
@@ -2,5 +2,16 @@
...
+        if args in cached_data:
```
Comment (vsmutok) [inline app/main.py]: Why you check only `args`?
What if function takes `kwargs`?

File: app/main.py [hunk 3]
```diff
@@ -2,5 +2,16 @@
...
+            print("Getting from cache")
+            return cached_data[args]
+
+        res = func(*args)
```
Comment (vsmutok) [inline app/main.py]: `res` not clear name

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (Timur5050) [reply app/main.py:8]: but the dict is not hashed, and there will be a TypeError when I try to use the dict as a key, how can I save it?

File: app/main.py [hunk 4]
```diff
@@ -1,6 +1,23 @@
...
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_data = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> any:
+        kwargs_value = tuple()
+        for key, value in kwargs.items():
+            kwargs_value += (key, value)
```
Comment (Dimosphen1) [inline app/main.py:12]: You could convert `kwargs` to string and store them in this way avoiding iteration

File: app/main.py [hunk 5]
```diff
@@ -1,6 +1,23 @@
...
 from typing import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_data = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> any:
```
Comment (Dimosphen1) [inline app/main.py:9]: It is preferable to use `Any` from `typing` since it is not deprecated

**Review by Dimosphen1 — APPROVED** [review]
LGTM, several comments were left for consideration

**Pull request closed without merging**